cmd/bootkube/render.go

```diff
@@ -112,12 +116,39 @@ func flagsToAssetConfig() (c *asset.Config, err error) {
 			return nil, err
 		}
 	}
+
+	_, podNet, err := net.ParseCIDR(renderOpts.podCIDR)
+	if err != nil {
+		return nil, err
+	}
+
+	_, serviceNet, err := net.ParseCIDR(renderOpts.serviceCIDR)
+	if err != nil {
+		return nil, err
+	}
+
+	if podNet.Contains(serviceNet.IP) || serviceNet.Contains(podNet.IP) {
+		return nil, fmt.Errorf("Pod CIDR %s and service CIDR %s must not overlap", podNet.String(), serviceNet.String())
+	}
+
+	dnsServiceIP, err := pickDNSServiceIP(renderOpts.serviceCIDR)
+	if err != nil {
+		return nil, err
+	}
+
+	if !renderOpts.selfHostKubelet {
+		fmt.Printf("Host kubelet MUST use --cluster-dns=%s\n", dnsServiceIP)
```

Comment:
It'll be shown every time since our clusters use host kubelet so I kinda viewed it as info. But I can see it being considered a warning.